Loop/Managers/LoopDataManager.swift

```diff
@@ -653,32 +653,35 @@ extension LoopDataManager {
         }
     }
 
-    func storeDosingDecision(withError error: Error? = nil) {
-        getLoopState { manager, state in
-            var dosingDecision = StoredDosingDecision()
-
-            dosingDecision.carbsOnBoard = state.carbsOnBoard
-            dosingDecision.predictedGlucose = state.predictedGlucose
-            dosingDecision.predictedGlucoseIncludingPendingInsulin = state.predictedGlucoseIncludingPendingInsulin
-            if let (recommendation: recommendation, date: date) = state.recommendedTempBasal {
-                dosingDecision.tempBasalRecommendationDate = TempBasalRecommendationDate(recommendation: recommendation, date: date)
-            }
-            dosingDecision.recommendedBolus = state.recommendedBolus?.recommendation.amount
-            if let lastReservoirValue = manager.doseStore.lastReservoirValue {
-                dosingDecision.lastReservoirValue = LastReservoirValue(startDate: lastReservoirValue.startDate, unitVolume: lastReservoirValue.unitVolume)
-            }
-            dosingDecision.pumpManagerStatus = self.delegate?.pumpManagerStatus
-            dosingDecision.error = error ?? state.error
+    func storeDosingDecision(withDate date: Date = Date(), withError error: Error? = nil) {
```

Comment:
Out of curiosity: is this under unit test already?

Reply:
There are already tests for storing a dosing decision.